external-crates/move/tools/move-package/src/resolution/dependency_graph.rs

```diff
@@ -1452,25 +1457,36 @@ fn deps_equal<'a>(
     graph1_pkg_table: &'a BTreeMap<PM::PackageName, Package>,
     graph2: &'a DependencyGraph,
     graph2_pkg_table: &'a BTreeMap<PM::PackageName, Package>,
+    overrides: &'a BTreeMap<PM::PackageName, Package>,
 ) -> std::result::Result<
     (),
     (
         Vec<(&'a Dependency, PM::PackageName, &'a Package)>,
         Vec<(&'a Dependency, PM::PackageName, &'a Package)>,
     ),
 > {
-    let graph1_edges = BTreeSet::from_iter(
-        graph1
-            .package_graph
-            .edges(pkg_name)
-            .map(|(_, pkg, dep)| (dep, pkg, graph1_pkg_table.get(&pkg).unwrap())),
-    );
-    let graph2_edges = BTreeSet::from_iter(
-        graph2
-            .package_graph
-            .edges(pkg_name)
-            .map(|(_, pkg, dep)| (dep, pkg, graph2_pkg_table.get(&pkg).unwrap())),
-    );
+    let graph1_edges =
+        BTreeSet::from_iter(graph1.package_graph.edges(pkg_name).map(|(_, pkg, dep)| {
+            (
+                dep,
+                pkg,
+                graph1_pkg_table
+                    .get(&pkg)
+                    .or_else(|| overrides.get(&pkg))
+                    .unwrap(),
```

Comment:
Rather than an `unwrap` can we return a result with an `ICE: ` message, so we fail a bit more gracefully (or can we make an argument here why this `unwrap` should never panic?)

Reply:
I added a comment on why these should not fail - if they do, it's a problem with an algorithm and the sooner someone reports this, the better.